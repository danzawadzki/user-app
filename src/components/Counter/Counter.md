Default counter component with all props:

```jsx
<Counter number={999} label="Label" />
```

Default counter component with out number prop (which is optional):

```jsx
<Counter label="Label" />
```
