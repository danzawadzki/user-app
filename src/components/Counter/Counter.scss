@import '../../assets/styles/colors.scss';

.Counter {
	display: flex;
	flex-direction: column;

	&__number {
		color: $brand--accent;
		font-size: 2.4rem;
		line-height: 2.9rem;
		letter-spacing: 0;
	}
	&__label {
		color: $blue--light;
		font-size: 1rem;
		line-height: 1.75rem;
		letter-spacing: 0;
	}
}
