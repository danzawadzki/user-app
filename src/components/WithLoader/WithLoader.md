Default WithLoader, where flag isLoading is true:

```jsx
const Test = WithLoader(Counter);
const label = 'Label';
const counter = 997;
<Test isLoading={true} label={label} number={counter} />;
```
