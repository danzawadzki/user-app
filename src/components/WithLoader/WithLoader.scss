@import '../../assets/styles/colors';

.WithLoader {
	display: flex;
	justify-content: center;
	align-items: center;
	min-height: 20rem;

	&__spinner {
		width: 40px;
		height: 40px;

		border: 2px solid $app__borders;
		border-top: 3px solid $brand--main;
		border-radius: 100%;

		animation: spin 1s infinite linear;
	}

	@keyframes spin {
		from {
			transform: rotate(0deg);
		}
		to {
			transform: rotate(360deg);
		}
	}
}
