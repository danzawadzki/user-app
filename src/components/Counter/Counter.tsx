import * as React from 'react';
import './Counter.scss';

export interface ICounter {
	/** Large counter number */
	number?: number;
	/** Small label text under the counter */
	label: string;
}

/**
 * Counter component.
 *
 * @version 1.0.0
 */
const Counter: React.StatelessComponent<ICounter> = ({ number = 0, label }) => (
	<div className="Counter">
		<span className="Counter__number">{number}</span>
		<span className="Counter__label">{label}</span>
	</div>
);

export default Counter;
