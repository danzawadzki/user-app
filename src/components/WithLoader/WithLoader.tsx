import * as React from 'react';
import './WithLoader.scss';

export interface IWithLoader {
	/** Loading flag */
	isLoading: boolean;
}

/**
 * WithLoader higher order component.
 *
 * @version 1.0.0
 */
const WithLoader = <P extends object>(Component: React.ComponentType<P>) =>
	class WithLoading extends React.Component<P & IWithLoader> {
		render() {
			const { isLoading, ...props } = this.props as IWithLoader;
			return isLoading ? (
				<div className="WithLoader">
					<div className="WithLoader__spinner" />
				</div>
			) : (
				<Component {...props} />
			);
		}
	};

export default WithLoader;
