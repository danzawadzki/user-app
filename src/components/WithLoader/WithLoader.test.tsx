import React from 'react';
import { shallow } from 'enzyme';
import WithLoader from './WithLoader';
import Counter from '../Counter/Counter';

describe('WithLoader component: ', () => {
	// Shallow rendered component
	const Test = WithLoader(Counter);
	const label = 'Label';
	const counter = 997;
	const component = shallow(<Test isLoading={true} label={label} number={counter} />);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should contain spinner if is loading', () => {
		expect(component.html()).not.toContain(label);
		expect(component.find('.WithLoader__spinner')).toBeTruthy();
	});

	it('should not contain spinner if is not loading', () => {
		const componentNotLoading = shallow(
			<Test isLoading={false} label={label} number={counter} />
		);
		expect(componentNotLoading.html()).toContain(label);
		expect(componentNotLoading.html()).toContain(counter);
	});
});
