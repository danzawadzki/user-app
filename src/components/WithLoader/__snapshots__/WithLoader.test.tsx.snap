// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component:  should render match snapshot 1`] = `
<div
  className="WithLoader"
>
  <div
    className="WithLoader__spinner"
  />
</div>
`;
